src/app/components/Carousel/Carousel.js

```diff
@@ -114,14 +111,39 @@ class Carousel extends Component {
         >
           <i className="fas fa-angle-left" />
         </button>
+
         <button
           type="button"
           className="carousel--button carousel--button-next"
           onClick={this.nextSlide}
-          disabled={data.length - scrollCounter === itemsPerView}
+          disabled={data.length - scrollCounter === visibleItems}
         >
           <i className="fas fa-angle-right" />
         </button>
+      </>
+    );
+  }
+
+  render() {
+    const { data, extended } = this.props;
+
+    return (
+      <div className="carousel--wrapper" ref={this.wrapperRef}>
+        <div
+          className={extended ? 'carousel' : 'wishlist-block'}
+          style={this.carouselStyleSheet}
+        >
+          {data &&
+            data.map(el => (
```

Comment:
don't set the list content inside carousel use instead `{this.props.children}`
const list = this.state.someArray.map(item => <AnyComponent />)
`<Carousel>  {list} /*children */ </Carousel>`